packages/sshnpdpy/lib/sshnpdclient.py

```diff
@@ -0,0 +1,318 @@
+from io import StringIO
+import logging
+import os, threading
+import subprocess
+from queue import Empty, Queue
+from time import sleep
+from uuid import uuid4
+from paramiko import AutoAddPolicy, SSHClient
+from paramiko.ed25519key import Ed25519Key
+
+
+from socket import socket
+from select import select
+
+from at_client import AtClient
+from at_client.common import AtSign
+from at_client.util import EncryptionUtil
+from at_client.common.keys import AtKey, Metadata, SharedKey
+from at_client.connections.notification.atevents import AtEvent, AtEventType
+
+#TODO: Figure out all the exception stuff later
+
+
+class SSHNPDClient:
+    #Current opened threads (TODO: clean up threads)
+    threads = []
+    
+    def __init__(self, atsign, manager_atsign, device, username, verbose=False ):
+        #AtClient Stuff
+        self.atsign = atsign
+        self.manager_atsign = manager_atsign
+        self.device = device
+        self.username = username
+        self.at_client = AtClient(AtSign(atsign), queue=Queue(maxsize=20), verbose=verbose)
+        self.device_namespace = f".{device}.sshnp"
+        self.authenticated = False
+        
+        #Logger
+        self.logger = logging.getLogger("sshnpd")
+        self.logger.setLevel((logging.DEBUG if verbose else logging.INFO))
+        self.logger.addHandler(logging.StreamHandler())
+        
+        #Directory Stuff
+        home_dir = ""
+        if os.name == "posix":  # Unix-based systems (Linux, macOS)
+            home_dir = os.path.expanduser("~")
+        elif os.name == "nt":  # Windows
+            home_dir = os.path.expanduser("~")
+        else:
+            raise NotImplementedError("Unsupported operating system")
+        self.ssh_path = f"{home_dir}/.ssh"
+        
+    def start(self):
+        username_key = SharedKey(
+        "username", AtSign(self.atsign), AtSign(self.manager_atsign))
+        self.at_client.put(username_key, self.username)
```

Comment:
This should only happen if the -u flag is supplied

Reply:
Made changes, will reach out to Gary today to talk about it.